### DISCUSSION TITLE
中英文清晰免费的字体推荐

### DISCUSSION THREAD

**Discussion by ZhuYajunFly** [BYJRK/DotNet-Discussions#338] [通用]
FontFamily="STSong" 宋体显示中文比较得体，但是显示字母和数字有些别扭，不清晰，不如FontFamily="Arial"，字体大小都是FontSize="18"，默认字重。
我经常把这个分成两个TextBox去显示，稍显麻烦。
如果碰到像 “10分钟”  “20分钟” 这种分不开的无奈使用FontFamily="STSong" ，导致中英文显示不统一。
我在网上看了很多字体，显示效果都不如FontFamily="STSong" 和 FontFamily="Arial"。
网上还有通过FontForge组合字体，这个没弄过。

![image](https://github.com/user-attachments/assets/327d61a7-f532-4edc-9bf1-e868a6355c26)
先指定英文字体，然后中文，能够生效的，不过英文和水平中线不一样。

**经过我这两天的测试 ，Arimo接近Arial，思源宋体 Light接近宋体，编程采用英文在前，中文在后比较接近需求。 TextElement.FontFamily="Arimo,思源宋体 Light"**
![image](https://github.com/user-attachments/assets/541dc26e-e343-4b76-8dae-2f63ca90834e)

**Comment by BYJRK** [comment]
你对英文的字体有什么要求吗？默认的比如微软雅黑，以及其他的一些国产的免费字体，比如 MiSans、鸿蒙字体、阿里普惠体之类的可以满足你的要求吗

**Reply by ZhuYajunFly** [reply]
> 你对英文的字体有什么要求吗？默认的比如微软雅黑，以及其他的一些国产的免费字体，比如 MiSans、鸿蒙字体、阿里普惠体之类的可以满足你的要求吗

很多都能满足，主要是一种字体能够达到中文宋体+英文及数字Arial的效果就好，测试了很多单独的字体，达不到想要的效果。

**Reply by ZhuYajunFly** [reply]
> 你对英文的字体有什么要求吗？默认的比如微软雅黑，以及其他的一些国产的免费字体，比如 MiSans、鸿蒙字体、阿里普惠体之类的可以满足你的要求吗

比如这样一段文字WPF FontSize="18" 显示 "12点整，航向12.78°，高度1234.68MM，温度36.78℃，湿度23.59%"，单独的一个字体显示效果不好。

**Reply by BYJRK** [reply]
你使用多个 TextBlock 或者 Run，然后指定不同的字体可以吗？如果不行的话，恐怕只能自己去制作一个字体文件了

**Reply by ZhuYajunFly** [reply]
> 你使用多个 TextBlock 或者 Run，然后指定不同的字体可以吗？如果不行的话，恐怕只能自己去制作一个字体文件了


 目前是多个TextBlock，确实有点麻烦。网上说可以单独指定中英文字体，不过我看实际效果，由于中文字体包含了英文及数字文本编码，给覆盖了。

**Reply by ZhuYajunFly** [reply]
> 你对英文的字体有什么要求吗？默认的比如微软雅黑，以及其他的一些国产的免费字体，比如 MiSans、鸿蒙字体、阿里普惠体之类的可以满足你的要求吗

![image](https://github.com/user-attachments/assets/1e7718ce-2b3c-43a0-a863-b560150c7395)
单一字体不尽人意
